frappe/www/printview.py

```diff
@@ -313,16 +314,18 @@ def validate_print_permission(doc):
 
 
 def validate_key(key, doc):
-	document_share_key = frappe.db.exists(
-		"Document Share Key",
-		{"reference_doctype": doc.doctype, "reference_docname": doc.name, "key": key},
-		cache=True,
-	)
-	if document_share_key:
-		if frappe.get_cached_doc("Document Share Key", document_share_key).is_expired():
+	try:
+		document_key_expiry = frappe.get_cached_value(
+			"Document Share Key",
+			{"reference_doctype": doc.doctype, "reference_docname": doc.name, "key": key},
+			"expires_on",
+		)
+		if is_expired(document_key_expiry):
 			raise frappe.exceptions.LinkExpiredError
 		else:
 			return
+	except frappe.DoesNotExistError:
```

Comment:
`get_cached_value` does not raise `DoesNotExistError` since https://github.com/frappe/frappe/pull/16448

You can either query the name as well, or use `get_cached_doc`

Reply:
Since this PR is for v13, and `get_cached_value` still raises `DoesNotExistError` in v13... I'll have to keep this change as it is... will update it in v14 PR.